### PR TITLE
Authentication of user using Devise

### PR DESCRIPTION
**Pull request by coderacheal**
In this pull request I have configured devise to 

- Register a new user.
- Login user with a combination of email and password.
- Hash passwords and store in the database.
- Ask for confirmation by email.
- Can reset password.

### CODE REVIEW

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @coderacheal and @frempongdev , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request :red_circle: :red_circle: :red_circle: 

You have submitted a project, that had an error when running your migration files to test your project so make sure you fix it.
image
 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @coderacheal, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>
required-changes-4

## Highlights
Devise Has Been Set Up Properly :heavy_check_mark: 
All Required Authentication Has Been Added :heavy_check_mark: 
Users Able To SignUp, SignIn, And SignOut Successfully :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._
### Due to insufficient files I will leave a comment here :red_circle: 
- [ ] In the readme file under the `Author section, kindly fix the link to your Twitter account. If you do not have one then kindly remove it
image

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
### Important :red_circle:
 - [ ] Going forward please use more descriptive branch names, as this is important for documentation purposes. A descriptive branch name would have been something like, `authentication` or `add-devise-authentication`

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  </div>
+  
+  <div class="field">
+    <%= f.label :bio %><br />
+    <%= f.text_field :bio %>
+  </div>
+  
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
```
[the remark this answers is not in the document]
Comment (Munsa1) [reply app/views/devise/registrations/new.html.erb:23]: Remember it should be for registration and login pages as per requirements [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md)

**Review by fobadara — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Clean code
✅ Good first attempt
 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Clean code
✅ Good first attempt
 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -1,26 +1,20 @@
...
-      <%= f.check_box :remember_me %>
-      <%= f.label :remember_me %>
-    </div>
-  <% end %>
+  <%# <% if devise_mapping.rememberable? %>
+    <%# <div class="field"> %>
+      <%# <%= f.check_box :remember_me %> 
+      <%# <%= f.label :remember_me %> 
+    <%# </div> %>
+  <%# <% end %>
```
Comment (CollinsTatang) [inline app/views/devise/sessions/new.html.erb:15]: ### **Optional**
- [ ] Please remove all commented code from your codebase to make your codebase clean and easy to maintain.

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- Your project meets all the requested changes by the previous code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**